src/Action/ExceptionAction.php

```diff
@@ -46,7 +49,7 @@ public function __construct(private readonly SerializerInterface $serializer, pr
     /**
      * Converts an exception to a JSON response.
      */
-    public function __invoke(FlattenException $exception, Request $request): Response
+    public function __invoke(FlattenException|ErrorExceptionInterface $exception, Request $request): Response
```

Comment:
```suggestion
    /**
     * @param FlattenException|ErrorExceptionInterface $exception
     */
    public function __invoke($exception, Request $request): Response
```

To avoid breaking if this class is used by users.

Reply:
Is this considered breaking to add a union type ? I'm also not sure how this class could be directly used by users ?
One problem with your suggestion is that in `ErrorListener::onControllerArgument`, I need to check the types of the controller argument (with Reflection) to transform the `\Throwable` in an instance of `ErrorExceptionInterface`.
I am definitely open to other possibilities though

Reply:
As long as a class  is not marked `@internal` you can't change a public method argument type (https://symfony.com/doc/current/contributing/code/bc.html). If we can't, you need to deprecate this controller and create a new one.